File: backend/src/main/java/fu/osms/inventory/service/StockDeliveryService.java
package fu.osms.inventory.service;

import fu.osms.inventory.dto.request.StockDeliveryRequest;
import fu.osms.inventory.dto.response.StockDeliveryResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.UUID;

public interface StockDeliveryService {

    /**
     * Create a new stock delivery document
     * @param request Stock delivery request
     * @return Created stock delivery response
     */
    StockDeliveryResponse createStockDelivery(StockDeliveryRequest request);

    /**
     * Update a draft stock delivery document
     * @param id Delivery ID
     * @param request Stock delivery request
     * @return Updated stock delivery response
     */
    StockDeliveryResponse updateStockDelivery(UUID id, StockDeliveryRequest request);

    /**
     * Get stock delivery by ID
     * @param id Delivery ID
     * @return Stock delivery response
     */
    StockDeliveryResponse getStockDeliveryById(UUID id);

    /**
     * Get all stock deliveries with filters and pagination
     * @param warehouseId Warehouse ID filter (optional)
     * @param status Status filter (optional)
     * @param deliveryType Delivery type filter (optional)
     * @param startDate Start date filter (optional)
     * @param endDate End date filter (optional)
     * @param pageable Pagination information
     * @return Page of stock deliveries
     */
    Page<StockDeliveryResponse> getAllStockDeliveries(
            UUID warehouseId,
            String status,
            String deliveryType,
            String keyword,
            LocalDate startDate,
            LocalDate endDate,
            Pageable pageable
    );

    /**
     * Confirm stock delivery (change status from DRAFT to CONFIRMED)
     * @param id Delivery ID
     * @return Updated stock delivery response
     */
    StockDeliveryResponse confirmStockDelivery(UUID id);

    /**
     * Cancel stock delivery
     * @param id Delivery ID
     * @return Updated stock delivery response
     */
    StockDeliveryResponse cancelStockDelivery(UUID id);

    /**
     * Get delivery statistics
     * @return Delivery statistics
     */
    Object getDeliveryStatistics();

    int syncPendingMarketplaceInventory();
}

File: frontend/src/features/inventory/services/stockDeliveryService.js
import stockDeliveryApi from '../../../api/stockDeliveryApi';

const stockDeliveryService = {
  // Create a new stock delivery
  createStockDelivery: async (deliveryData) => {
    try {
      const response = await stockDeliveryApi.createStockDelivery(deliveryData);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Update draft stock delivery
  updateStockDelivery: async (deliveryId, deliveryData) => {
    try {
      const response = await stockDeliveryApi.updateStockDelivery(deliveryId, deliveryData);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get stock delivery by ID
  getStockDeliveryById: async (deliveryId) => {
    try {
      const response = await stockDeliveryApi.getStockDeliveryById(deliveryId);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get all stock deliveries with filters and pagination
  getAllStockDeliveries: async (filters = {}) => {
    try {
      const response = await stockDeliveryApi.getAllStockDeliveries(filters);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Confirm stock delivery
  confirmStockDelivery: async (deliveryId) => {
    try {
      const response = await stockDeliveryApi.confirmStockDelivery(deliveryId);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Cancel stock delivery
  cancelStockDelivery: async (deliveryId) => {
    try {
      const response = await stockDeliveryApi.cancelStockDelivery(deliveryId);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get delivery statistics
  getDeliveryStatistics: async () => {
    try {
      const response = await stockDeliveryApi.getDeliveryStatistics();
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  syncPendingMarketplaceInventory: async () => {
    try {
      const response = await stockDeliveryApi.syncPendingMarketplaceInventory();
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  getOrderCandidates: async (params = {}) => {
    try {
      const response = await stockDeliveryApi.getOrderCandidates(params);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  createFromOrders: async (orderIds) => {
    try {
      const response = await stockDeliveryApi.createFromOrders(orderIds);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },
};

export default stockDeliveryService;
